genetIC/src/simulation/particles/zeldovich.hpp

```diff
@@ -202,26 +202,40 @@ namespace particle {
 
       // Differentiation operator
       auto differentiate = [a, b, &potential](auto & zeldovichOffsetField, int dir) {
-        Coordinate<int> directionVector;
-        Coordinate<int> negDirectionVector;
-
         zeldovichOffsetField->toReal();
-        directionVector[dir] = 1;
-        negDirectionVector[dir] = -1;
 
         // Create zeldovich offset field
         std::vector<GridDataType> &data = zeldovichOffsetField->getDataVector();
         auto grid2 = zeldovichOffsetField->getGrid();
 
+        // Cannot compute second order finite-difference on a grid smaller than 2
+        assert(grid2.size >= 2);
+
         // Iterate over cells and compute finite difference
         grid2.parallelIterateOverCellsSpatiallyClustered(
-           [&data, grid2, &potential, a, b, directionVector, negDirectionVector](size_t index){
+           [&data, grid2, &potential, a, b, dir](size_t index){
```

Comment:
While re-reading this, I am wondering whether this shouldn't be instead
```suggestion
           [&data, &grid2, &potential, a, b, dir](size_t index){
```

Reply:
What is your opinion @apontzen ?

Reply:
I don't think it's a big deal either way since the copy is very cheap. The advantage of capturing value rather than reference is that you know at a glance the inner part won't change `grid2`. Then again, I think most methods of Grid are const anyway.

Reply:
OK! That's the way it is implemented as of 34c2806 .